docs/preview/02-Guidance/Service-to-service Correlation/use-service-to-service-correlation-in-web-api.md

```diff
@@ -0,0 +1,310 @@
+[ApiController]
+[Route("/api/v1/stock")]
+public class StockController : ControllerBase
+{
+    [HttpGet]
+    public async Task<IActionResult> Get([FromQuery] string productName)
+    {
+        var response = new ProductStockResponse
+        {
+            ProductName = productName,
+            AvailableItems = 3
+        };
+
+        return Ok(response);
+    }
+}
+```
+
+Right now, we haven't used any Arcus functionality, only built-in ASP.NET Core features.
+
+### Product API: add Arcus HTTP correlation
+Now that we have explained the application code, we can add the Arcus functionality that will provide clear telemetry for the API interaction.
+
+First, these packages need to be installed:
+```shell
+PM > Install-Package Arcus.WebApi.Logging -MinimumVersion 1.6.0
+PM > Install-Package Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights -MinimumVersion 2.5.0
+```
+
+> For more information on `Arcus.WebApi.Logging`, see [these dedicated feature docs](https://webapi.arcus-azure.net/features/logging), for more information on `Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights`, see [these dedicated feature docs](https://observability.arcus-azure.net/Features/sinks/azure-application-insights).
+
+The following additions has to be made to the startup code for the **Product API** to be able to track the incoming/outgoing HTTP requests, and to log to Application Insights.
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
+        builder.Services.AddRouting();
+        builder.Services.AddControllers();
+        
+        // [Add] Makes HTTP correlation available throughout application
+        builder.Services.AddHttpCorrelation();
+
+        builder.Configuration.AddInMemoryCollection(new[]
+        {
+            new KeyValuePair<string, string>("STOCK_API_URL", "http://localhost:788")
+        });
+        builder.Services.AddHttpClient("Stock API")
+                        // [Add] Adds telemetry to outgoing request
+                        .WithHttpCorrelationTracking();
+
+        // [Add] Serilog configuration that writes to Application Insights
+        builder.Host.UseSerilog((context, provider, config) =>
+        {
+            config.MinimumLevel.Verbose()
+                  .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
+                  .Enrich.WithComponentName("Product API")
+                  .Enrich.WithHttpCorrelationInfo(provider)
+                  .WriteTo.AzureApplicationInsightsWithConnectionString("<connection-string>");
+        });
+
+        WebApplication app = builder.Build();
+        // [Add] Gets HTTP correlation from incoming request
+        app.UseHttpCorrelation();
+        app.UseRouting();
+        // [Add] Tracks incoming request
+        app.UseRequestTracking();
```

Comment:
I was wondering if the order of adding these middleware components is important.   If so, we should stress that.

Reply:
Yes, it is important, and is also mentioned in the Web API feature docs, but yes, we could duplicate this warning here.

Reply:
Added in https://github.com/arcus-azure/arcus.observability/pull/446/commits/bd1f6f5ed4baea39c56bfa7c940bb98169f482b4